File: src/test/java/org/cloudfoundry/credhub/interceptor/UserContextInterceptorTest.java
package org.cloudfoundry.credhub.interceptor;

import javax.servlet.http.HttpServletRequest;

import org.springframework.mock.web.MockHttpServletResponse;
import org.springframework.security.core.Authentication;

import org.cloudfoundry.credhub.auth.UserContext;
import org.cloudfoundry.credhub.auth.UserContextFactory;
import org.cloudfoundry.credhub.auth.UserContextHolder;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static junit.framework.TestCase.assertFalse;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(JUnit4.class)
public class UserContextInterceptorTest {
  private UserContextInterceptor subject;
  private UserContext userContext;
  private UserContextFactory userContextFactory;
  private UserContextHolder userContextHolder;
  private HttpServletRequest request;
  private MockHttpServletResponse response;

  @Before
  public void setup() {
    userContextFactory = mock(UserContextFactory.class);
    userContext = mock(UserContext.class);
    userContextHolder = new UserContextHolder();

    subject = new UserContextInterceptor(userContextFactory, userContextHolder);

    when(userContextFactory.createUserContext(any())).thenReturn(userContext);
    request = mock(HttpServletRequest.class);
    response = new MockHttpServletResponse();
  }

  @Test
  public void preHandle_setsUserContextFromPrincipal() throws Exception {
    when(request.getUserPrincipal()).thenReturn(mock(Authentication.class));
    subject.preHandle(request, response, new Object());

    assertThat(userContextHolder.getUserContext(), equalTo(userContext));
  }

  @Test
  public void preHandle_ReturnsFalseWhenNoPrincipal() throws Exception {
    when(request.getUserPrincipal()).thenReturn(null);
    final boolean result = subject.preHandle(request, response, new Object());
    assertFalse(result);
  }
}

File: components/auth/src/test/kotlin/org/cloudfoundry/credhub/interceptors/UserContextInterceptorTest.kt
package org.cloudfoundry.credhub.interceptors

import jakarta.servlet.http.HttpServletRequest
import org.cloudfoundry.credhub.auth.UserContext
import org.cloudfoundry.credhub.auth.UserContextFactory
import org.cloudfoundry.credhub.auth.UserContextHolder
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.ArgumentMatchers.any
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`
import org.springframework.mock.web.MockHttpServletResponse
import org.springframework.security.core.Authentication
import java.security.Principal

class UserContextInterceptorTest {
    private var subject: UserContextInterceptor? = null
    private var userContext: UserContext? = null
    private var userContextFactory: UserContextFactory? = null
    private var userContextHolder: UserContextHolder? = null
    private var request: HttpServletRequest? = null
    private var response: MockHttpServletResponse? = null

    @BeforeEach
    fun setup() {
        userContextFactory = mock(UserContextFactory::class.java)
        userContext = mock(UserContext::class.java)
        userContextHolder = UserContextHolder()

        subject = UserContextInterceptor(userContextFactory!!, userContextHolder!!)

        `when`(userContextFactory!!.createUserContext(any<Authentication>())).thenReturn(userContext)
        request = mock(HttpServletRequest::class.java)
        response = MockHttpServletResponse()
    }

    @Test
    @Throws(Exception::class)
    fun preHandle_setsUserContextFromPrincipal() {
        `when`<Principal>(request!!.userPrincipal).thenReturn(mock(Authentication::class.java))
        subject!!.preHandle(request!!, response!!, Any())

        assertThat<UserContext>(userContextHolder!!.userContext, equalTo<UserContext>(userContext))
    }

    @Test
    @Throws(Exception::class)
    fun preHandle_ReturnsFalseWhenNoPrincipal() {
        `when`<Principal>(request!!.userPrincipal).thenReturn(null)
        val result = subject!!.preHandle(request!!, response!!, Any())
        assertFalse(result)
    }
}
